kahuna/public/js/search-query/service.js

```diff
@@ -0,0 +1,25 @@
+import angular from 'angular';
+import {mapFactory} from '../lib/data-structure/map-factory';
+
+export const searchQueryService = angular.module('searchQuery.service', [
+    mapFactory.name
+]);
+
+searchQueryService.factory('searchQueryService', ['$state', '$stateParams', 'mapFactory',
```

Comment:
This is the other option to what is spoken about in [routes holding app state](https://github.com/guardian/grid/pull/1641/files#r49831719). Here a service holds the state, which then allows [embedded controllers to access it as a dependency](https://github.com/guardian/grid/pull/1641/files#diff-32fea343d36b21c11521e08ea0059cafR37).